src/Panel/Controller/Search.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Kirby\Panel\Controller;
+
+use Kirby\Cms\App;
+use Kirby\Toolkit\Escape;
+
+/**
+ * The Search controller takes care of the logic
+ * for delivering Panel search results
+ *
+ * @package   Kirby Panel
+ * @author    Nico Hoffmann <nico@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://getkirby.com/license
+ * @internal
+ */
+class Search
+{
+	public static function files(
+		string $query,
+		int|null $limit = null,
+		int $page = 1
+	): array {
+		$kirby = App::instance();
+		$files = $kirby->site()
+			->index(true)
+			->filter('isListable', true)
+			->files();
+
+		// add site files which aren't considered by the index
+		$files = $files->add($kirby->site()->files());
+
+		// filter and search among those files
+		$files = $files->filter('isListable', true)->search($query);
```

Comment:
Maybe we can write like that

```suggestion
		$files = $kirby->site()
			->index(true)
			->files()
			//add site files which aren't considered by the index
			->add($kirby->site()->files())
			->filter('isListable', true)
			->search($query);
```